onnxruntime/core/optimizer/constant_folding.cc

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/optimizer/constant_folding.h"
+#include "core/graph/graph_utils.h"
+#include "core/optimizer/optimizer_execution_frame.h"
+#include "core/framework/op_kernel.h"
+#include "core/framework/ml_value.h"
+
+namespace onnxruntime {
+
+Status ConstantFolding::Apply(Graph& graph, Node& node, bool& modified, bool& deleted) {
+  // TODO Check if we need default transformers any more. I dont think we do...
+
+  // Create execution frame for executing constant nodes.
+  OptimizerExecutionFrame::Info info({&node}, graph.GetAllInitializedTensors());
```

Comment:
It is not necessary to pass all of graph initializers. Instead, only initializers for this node is enough.

Reply:
That's what I had done initially, but it is more efficient this way.
The graph initializers are used to build the internal info structure, so only those needed are accessed (we don't copy everything or something). Otherwise, I would have to pay each time the cost of going all node's input defs, checking which are initializers, and adding them to a new set.

Reply:
Info ctor creates mlvalues based on the initializers set which is passed in. Thus, the cost will be high in Info ctor If you pass in all of graph initializers.

Reply:
Just double check the current implementation, it is OK to pass in all graph initializers since inside info ctor, the instances of mlvalue are only created for initializers of passing nodes.  Thanks!